packages/nextra/src/server/mdx-plugins/rehype-mathjax-react.ts

```diff
@@ -0,0 +1,88 @@
+import type { Element, Root } from 'hast'
+import { toText } from 'hast-util-to-text'
+import { SKIP, visitParents } from 'unist-util-visit-parents'
+import type { VFile } from 'vfile'
+
+/** @type {Readonly<Options>} */
+const emptyOptions = {}
+/** @type {ReadonlyArray<unknown>} */
+const emptyClasses: string[] = []
+
+export type Options = Record<string, unknown> | undefined
+
+/**
+ * Render elements with a `language-math` (or `math-display`, `math-inline`)
+ * class with KaTeX.
+ *
+ * @param {Readonly<Options> | null | undefined} [options]
+ *   Configuration (optional).
+ * @returns
+ *   Transform.
+ */
+export function rehypeReactMathjax(options: Options) {
+  const _settings = options || emptyOptions
+
+  /**
+   * Transform.
+   */
+  return function (tree: Root, _file: VFile) {
+    visitParents(tree, 'element', function (element, parents) {
+      const classes = Array.isArray(element.properties.className)
+        ? element.properties.className
+        : emptyClasses
+      // This class can be generated from markdown with ` ```math `.
+      const languageMath = classes.includes('language-math')
+      // This class is used by `remark-math` for flow math (block, `$$\nmath\n$$`).
```

Comment:
I think support of $$ was removed in remark-math v6? https://github.com/shuding/nextra/issues/2368#issuecomment-1747752925

Reply:
I think it would be good to still support this since someone may modify the parser and insert `display-math` tags manually.

Reply:
let's just use `rehype-mathjax` instead of copy pasting similar things, additional logic we can leave for our new plugin